src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py

```diff
@@ -0,0 +1,149 @@
+
+import math
+import torch
+from torch import Generator, Tensor, device as Device, dtype as Dtype, float32, tensor
+
+from refiners.foundationals.latent_diffusion.solvers.solver import NoiseSchedule, Solver
+
+# Copied from diffusers.schedulers.scheduling_ddpm.betas_for_alpha_bar
+def betas_for_alpha_bar(
+    num_diffusion_timesteps,
+    max_beta=0.999,
+    alpha_transform_type="cosine",
+):
+    """
+    Create a beta schedule that discretizes the given alpha_t_bar function, which defines the cumulative product of
+    (1-beta) over time from t = [0,1].
+
+    Contains a function alpha_bar that takes an argument t and transforms it to the cumulative product of (1-beta) up
+    to that part of the diffusion process.
+
+
+    Args:
+        num_diffusion_timesteps (`int`): the number of betas to produce.
+        max_beta (`float`): the maximum beta to use; use values lower than 1 to
+                     prevent singularities.
+        alpha_transform_type (`str`, *optional*, default to `cosine`): the type of noise schedule for alpha_bar.
+                     Choose from `cosine` or `exp`
+
+    Returns:
+        betas (`np.ndarray`): the betas used by the scheduler to step the model outputs
+    """
+    if alpha_transform_type == "cosine":
+
+        def alpha_bar_fn(t):
+            return math.cos((t + 0.008) / 1.008 * math.pi / 2) ** 2
+
+    elif alpha_transform_type == "exp":
+
+        def alpha_bar_fn(t):
+            return math.exp(t * -12.0)
+
+    else:
+        raise ValueError(f"Unsupported alpha_tranform_type: {alpha_transform_type}")
+
+    betas = []
+    for i in range(num_diffusion_timesteps):
+        t1 = i / num_diffusion_timesteps
+        t2 = (i + 1) / num_diffusion_timesteps
+        betas.append(min(1 - alpha_bar_fn(t2) / alpha_bar_fn(t1), max_beta))
+    return torch.tensor(betas, dtype=torch.float32)
+
+class ConsistencyDecoderSolver(Solver):
```

Comment:
Is it possible to use instead the existing EulerSolver? If not, what are the differences?

Reply:
The ConsistencyDecoderVAE comes with a custom, specific scheduler class in diffusers.
I'll double check to see if it boils down to Euler with well chosen inference parameters.

Reply:
Thanks. In any case, you should implement ConsistencyDecoder using Refiners terminology for solvers (we don't use greek letters)

Reply:
:wave: note that we changed the interface of solvers a bit in 0.4 (https://github.com/finegrain-ai/refiners/pull/305) so you will need to rebase and change a few things.

Reply:
Hi, 

Sorry for the wait, had a week off, I did the rebase. 

Will re-work the solver and do the e2e tests asap